### PR TITLE
[NCL-5167] Add endpoints to get Builds related to an artifact

### PR DESCRIPTION
**Pull request by janinko**
### Checklist:

* [x] Have you added unit tests for your change?

### CODE REVIEW

**Comment by janinko** [comment]
retest this please

File: spi/src/main/java/org/jboss/pnc/spi/datastore/predicates/BuildRecordPredicates.java [hunk 1]
```diff
@@ -172,6 +172,20 @@
...
+    public static Predicate<BuildRecord> withArtifactProduced(Integer artifactId) {
+        return (root, query, cb) -> {
+            SetJoin<BuildRecord, Artifact> builtArtifacts = root.join(BuildRecord_.builtArtifacts);
+            return cb.equal(builtArtifacts.get(Artifact_.id), artifactId);
+        };
+    }
+
+    public static Predicate<BuildRecord> withArtifactDependency(Integer artifactId) {
+        return (root, query, cb) -> {
+            SetJoin<BuildRecord, Artifact> builtArtifacts = root.join(BuildRecord_.dependencies);
```
Comment (michalovjan) [inline spi/src/main/java/org/jboss/pnc/spi/datastore/predicates/BuildRecordPredicates.java]: copy-paste error
builtArtifacts -> dependencies

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ArtifactEndpoint.java [hunk 2]
```diff
@@ -125,4 +129,34 @@ Artifact getSpecific(
...
+                    content = @Content(schema = @Schema(implementation = SwaggerPages.BuildPage.class))),
+                @ApiResponse(responseCode = INVALID_CODE, description = INVALID_DESCRIPTION,
+                    content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                @ApiResponse(responseCode = SERVER_ERROR_CODE, description = SERVER_ERROR_DESCRIPTION,
+                    content = @Content(schema = @Schema(implementation = ErrorResponse.class)))
+    })
+    @GET
+    @Path("/{id}/builds")
+    Page<Build> getBuilds(
+            @Parameter(description = GB_ID) @PathParam("id") String id,
```
Comment (michalovjan) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ArtifactEndpoint.java]: group-build id?

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ArtifactEndpoint.java [hunk 3]
```diff
@@ -125,4 +129,34 @@ Artifact getSpecific(
...
+                    content = @Content(schema = @Schema(implementation = SwaggerPages.BuildPage.class))),
+                @ApiResponse(responseCode = INVALID_CODE, description = INVALID_DESCRIPTION,
+                    content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                @ApiResponse(responseCode = SERVER_ERROR_CODE, description = SERVER_ERROR_DESCRIPTION,
+                    content = @Content(schema = @Schema(implementation = ErrorResponse.class)))
+    })
+    @GET
+    @Path("/{id}/dependant-builds")
+    Page<Build> getDependantBuilds(
+            @Parameter(description = GB_ID) @PathParam("id") String id,
```
Comment (michalovjan) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ArtifactEndpoint.java]: group-build id?

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/ArtifactEndpointTest.java [hunk 4]
```diff
@@ -240,4 +244,26 @@ public void shouldUpdateArtifact() throws ClientException {
...
+                .hasSize(1);
+    }
+
+    @Test
+    public void shouldGetBuildsThatDependsOnArtifact() throws RemoteResourceException {
+        ArtifactClient client = new ArtifactClient(RestClientConfiguration.getConfiguration(RestClientConfiguration.AuthenticateAs.USER));
+
+        RemoteCollection<Build> builds = client.getDependantBuilds(artifactRest3.getId());
+
+        Build a;
```
Comment (michalovjan) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/ArtifactEndpointTest.java]: not used variable

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/ArtifactEndpointTest.java [hunk 5]
```diff
@@ -240,4 +244,26 @@ public void shouldUpdateArtifact() throws ClientException {
...
         assertThat(artifact2.getSize()).isEqualTo(size);
     }
+
+    @Test
+    public void shouldGetBuildThatProducedArtifact() throws RemoteResourceException {
+        ArtifactClient client = new ArtifactClient(RestClientConfiguration.getConfiguration(RestClientConfiguration.AuthenticateAs.USER));
+
+        RemoteCollection<Build> builds = client.getBuilds(artifactRest1.getId());
+
+        Build a;
```
Comment (michalovjan) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/ArtifactEndpointTest.java]: not used variable

**Comment by janinko** [comment]
retest this please

**Pull request merged**